### PR TITLE
[Chore] Use helpers to simplify expression handling code

### PR DESCRIPTION
**Pull request by scovich**
Kernel provides quite a few helper methods that make it easier to create expressions, but not all call sites take advantage of them. Fix that.

Also update the various binary operation helpers to accept `impl Into<Expression>` as their RHS arg, which allows further simplifications.

### CODE REVIEW

File: kernel/src/expressions/mod.rs [hunk 1]
```diff
@@ -134,6 +134,13 @@ pub enum Expression {
     Column(String),
     /// A struct computed from a Vec of expressions
     Struct(Vec<Expression>),
+    /// A unary operation.
+    UnaryOperation {
```
Comment (scovich) [inline kernel/src/expressions/mod.rs:138]: Seems more logical to order these unary -> binary -> variadic.

File: kernel/src/engine/arrow_expression.rs [hunk 2]
```diff
@@ -609,23 +590,23 @@ mod tests {
         let batch = RecordBatch::try_new(Arc::new(schema.clone()), vec![Arc::new(values)]).unwrap();
         let column = Expression::column("a");
 
-        let expression = Box::new(column.clone().add(Expression::Literal(Scalar::Integer(1))));
```
Comment (scovich) [inline kernel/src/engine/arrow_expression.rs:612]: Not sure why these tests were boxing everything... but it's not needed now.

File: kernel/src/expressions/mod.rs [hunk 3]
```diff
@@ -234,7 +234,7 @@ impl Expression {
     }
 
     /// Create a new struct expression
-    pub fn struct_expr(exprs: impl IntoIterator<Item = Self>) -> Self {
+    pub fn struct_from(exprs: impl IntoIterator<Item = Self>) -> Self {
```
Comment (zachschuermann) [inline kernel/src/expressions/mod.rs:237]: flagging breaking change

**Review by zachschuermann — APPROVED** [review]
very nice! love the literal's From ergonomics :)

**Review by nicklan — APPROVED** [review]
lgtm!

**Pull request merged**